pulp_container/app/models.py

```diff
@@ -138,10 +140,17 @@ def init_annotations(self, manifest_data=None):
 
     def init_labels(self):
         if self.config_blob:
-            config_artifact = self.config_blob._artifacts.get()
-
-            config_data, _ = get_content_data(config_artifact)
-            self.labels = config_data.get("config", {}).get("Labels") or {}
+            try:
+                # if config-blob is stored in the database
+                config = json.loads(self.config_blob.config_blob.instance.data)["config"]
+                if "Labels" in config.keys():
+                    self.labels = config["Labels"] or {}
+            except json.decoder.JSONDecodeError:
+                # if not, we'll retrieve the config from file
+                config_artifact = self.config_blob._artifacts.get()
```

Comment:
I just realized. I know the chances are small, but this will fail if the image is after reclaim_space. What should we do?
a) nothing, we will not initialize labels because file is unavailable
b) fetch the file

I tend to go with a) because if space was intentionally reclaimed then it would be a surprise for the user to again see that data occupying the space.

Reply:
same observation stands for the manifest._artifacts.get() in the init_annotations...unrelated to this PR but while we are here...

Reply:
> I tend to go with a) because if space was intentionally reclaimed then it would be a surprise for the user to again see that data occupying the space.

I agree with this.